Linting error on null in dependencies is misleading

When doing `helm lint` on a chart with the following dependencies: 

```
dependencies:
  - name: foo
    version: 1.2.3
  - null
```

it fails with 
```
==> Linting .
[ERROR] templates/: validation: dependency cannot be an empty list
```

This is quite misleading, as a chart with an empty list (`dependencies: []`) does not fail linting.

This was caused by #11044 and I would suggest to simply change the error message to a more meaningful one. 

Output of `helm version`: 
```
version.BuildInfo{Version:"v3.10.2", GitCommit:"50f003e5ee8704ec937a756c646870227d7c8b58", GitTreeState:"clean", GoVersion:"go1.19.3"}
```

--- a/pkg/chart/dependency.go
+++ b/pkg/chart/dependency.go
@@ -54,7 +54,7 @@ type Dependency struct {
 // loaded.
 func (d *Dependency) Validate() error {
 	if d == nil {
-		return ValidationError("dependency cannot be an empty list")
+		return ValidationError("dependencies must not contain empty or null nodes")
 	}
 	d.Name = sanitizeString(d.Name)
 	d.Version = sanitizeString(d.Version)

--- a/pkg/chart/metadata.go
+++ b/pkg/chart/metadata.go
@@ -35,7 +35,7 @@ type Maintainer struct {
 // Validate checks valid data and sanitizes string characters.
 func (m *Maintainer) Validate() error {
 	if m == nil {
-		return ValidationError("maintainer cannot be an empty list")
+		return ValidationError("maintainers must not contain empty or null nodes")
 	}
 	m.Name = sanitizeString(m.Name)
 	m.Email = sanitizeString(m.Email)

--- a/pkg/chart/metadata_test.go
+++ b/pkg/chart/metadata_test.go
@@ -82,7 +82,7 @@ func TestValidate(t *testing.T) {
 					nil,
 				},
 			},
-			ValidationError("dependency cannot be an empty list"),
+			ValidationError("dependencies must not contain empty or null nodes"),
 		},
 		{
 			&Metadata{
@@ -94,7 +94,7 @@ func TestValidate(t *testing.T) {
 					nil,
 				},
 			},
-			ValidationError("maintainer cannot be an empty list"),
+			ValidationError("maintainers must not contain empty or null nodes"),
 		},
 		{
 			&Metadata{APIVersion: "v2", Name: "test", Version: "1.2.3.4"},

